### ISSUE TITLE
SDK initialisation and installation not working

### ISSUE THREAD

**Issue by kavathiyamahesh** [infobip/mobile-messaging-sdk-android#11]
Hello

I was facing issue while initializing infobip and adding custom attributes, Please check below logs.

```
V/MobileMessaging: UPDATE INSTALLATION >>>
D/MMHTTP: REQUEST: Request(httpMethod=PATCH, uri=https://mobile.infobip.com/mobile/1/appinstance/362c89db-0b6d-4683-b403-d80237be2de6, apiKey=<key>, credentials=null, headers={applicationcode=[5142d8416a], pushregistrationid=[362c89db-0b6d-4683-b403-d80237be2de6], foreground=[true]}, queryParams={}, body={regEnabled=true, customAttributes={transactions-notifications=true, service-notification=true}})
I/System.out: (HTTPLog)-Static: isSBSettingEnabled false
I/System.out: (HTTPLog)-Static: isSBSettingEnabled false
W/MMHTTP: Got SSL handshake exception javax.net.ssl.SSLHandshakeException: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.
W/MMHTTP: Will re-try in untrusted mode
V/MobileMessaging: Result wrapper: 
    {
      "cancelled": false,
      "error": {
        "code": "-4",
        "cause": {
          "detailMessage": "Cannot set request property after connection is made",
          "stackTrace": [],
          "suppressedExceptions": []
        },
        "detailMessage": "Can\u0027t access URI : https://mobile.infobip.com/mobile/1/appinstance/362c89db-0b6d-4683-b403-d80237be2de6",
        "stackTrace": [],
        "suppressedExceptions": []
      },
      "inputs": []
    }
V/MobileMessaging: UPDATE INSTALLATION ERROR <<<
    org.infobip.mobile.messaging.api.support.ApiIOException: -4, Can't access URI : https://mobile.infobip.com/mobile/1/appinstance/362c89db-0b6d-4683-b403-d80237be2de6
        at org.infobip.mobile.messaging.api.support.http.client.DefaultApiClient.execute(DefaultApiClient.java:104)
        at org.infobip.mobile.messaging.api.support.Generator$CachingInvocationHandler.invoke(Generator.java:368)
        at java.lang.reflect.Proxy.invoke(Proxy.java:393)
        at $Proxy2.patchInstance(Unknown Source)
        at org.infobip.mobile.messaging.mobile.appinstance.InstallationSynchronizer$2.run(InstallationSynchronizer.java:209)
        at org.infobip.mobile.messaging.mobile.appinstance.InstallationSynchronizer$2.run(InstallationSynchronizer.java:205)
        at org.infobip.mobile.messaging.mobile.common.MRetryableTask$1.run(MRetryableTask.java:82)
        at org.infobip.mobile.messaging.mobile.common.MAsyncTask$2.doInBackground(MAsyncTask.java:54)
        at org.infobip.mobile.messaging.mobile.common.MAsyncTask$2.doInBackground(MAsyncTask.java:42)
        at android.os.AsyncTask$2.call(AsyncTask.java:295)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
        at java.lang.Thread.run(Thread.java:818)
     Caused by: java.lang.IllegalStateException: Cannot set request property after connection is made
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.setRequestProperty(HttpURLConnectionImpl.java:545)
        at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.setRequestProperty(DelegatingHttpsURLConnection.java:258)
        at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.setRequestProperty(HttpsURLConnectionImpl.java:25)
        at org.infobip.mobile.messaging.api.support.http.client.DefaultApiClient.executeHTTP(DefaultApiClient.java:119)
        at org.infobip.mobile.messaging.api.support.http.client.DefaultApiClient.execute(DefaultApiClient.java:94)
        at org.infobip.mobile.messaging.api.support.Generator$CachingInvocationHandler.invoke(Generator.java:368) 
        at java.lang.reflect.Proxy.invoke(Proxy.java:393) 
        at $Proxy2.patchInstance(Unknown Source) 
        at org.infobip.mobile.messaging.mobile.appinstance.InstallationSynchronizer$2.run(InstallationSynchronizer.java:209) 
        at org.infobip.mobile.messaging.mobile.appinstance.InstallationSynchronizer$2.run(InstallationSynchronizer.java:205) 
        at org.infobip.mobile.messaging.mobile.common.MRetryableTask$1.run(MRetryableTask.java:82) 
        at org.infobip.mobile.messaging.mobile.common.MAsyncTask$2.doInBackground(MAsyncTask.java:54) 
        at org.infobip.mobile.messaging.mobile.common.MAsyncTask$2.doInBackground(MAsyncTask.java:42) 
        at android.os.AsyncTask$2.call(AsyncTask.java:295) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:237) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) 
        at java.lang.Thread.run(Thread.java:818)

```

I am facing this issue before One hour 12:30PM-1: 00 PM IST, I have tried with multiple devices and facing the same issues. 

Again I have tried after 30 min and it's working fine without changes any piece of code. so May I know what was the reasons that I am facing above issue for a particular time?

Thanks

**Comment by kavathiyamahesh** [comment]
Hello @stanislavin @tjuric 

I am again facing this issue since from 28 MARCH, can you please check this, why I can't able initialize or activate notification on a specific device like Meiuze PRO 7

**Comment by tjuric** [comment]
@kavathiyamahesh we'll check it up and update you on our findings.

**Comment by stanislavin** [comment]
Should be fixed in 2.0.12. Related to improper connection usage when using untrusted mode.

**Comment by kavathiyamahesh** [comment]
Hello @stanislavin 
Thanks for the update, looks like the issue is fixed.
